src/pos.cpp

```diff
@@ -97,6 +97,44 @@ double GetPoSKernelPS()
     return result;
 }
 
+double GetPoSKernelPS(int nHeight)
+{
+    int nPoSInterval = 72;
+    double dStakeKernelsTriedAvg = 0;
+    int nStakesHandled = 0, nStakesTime = 0;
+
+    CBlockIndex *pindex = ChainActive()[nHeight];
```

Comment:
It might break if you pass -1 here or >MaxHeight ?

Reply:
If we call somewhere else, then it will be necessary to do the check at the top again.